### PR TITLE
Attempt to be more flexible about parsing Semtech JSON packets

### PR DESCRIPTION
**Pull request by Vagabond**
Additionally fix an edge case when no gateways are connected and also
fix a bug in the fake radio backplane.

### CODE REVIEW

**Comment by vihu** [comment]
There are a lot of `maps:get/2` calls here which I'm assuming are meant to blow up if the key isn't found?

File: src/miner_lora.erl [hunk 1]
```diff
@@ -151,9 +155,14 @@ mk_token(Timers) ->
...
 
--spec select_gateway(map()) -> gateway().
+-spec select_gateway(map()) -> {ok, gateway()} | {error, no_gateways}.
 select_gateway(Gateways) ->
-    erlang:element(2, erlang:hd(maps:to_list(Gateways))).
+    case maps:size(Gateways) of
+        0 ->
+            {error, no_gateways};
+        _ ->
+            {ok, erlang:element(2, erlang:hd(maps:to_list(Gateways)))}
```
Comment (vihu) [inline src/miner_lora.erl]: Select the first gateway always? Can we add a comment here if we need to update this down the line?

**Comment by Vagabond** [comment]
> There are a lot of `maps:get/2` calls here which I'm assuming are meant to blow up if the key isn't found?
Yes, they're mandatory fields

File: src/miner_lora.erl (hunk 1 above)
Comment (Vagabond) [reply src/miner_lora.erl]: yeah, this will change with the 'multi tenant' miner

Comment (Vagabond) [reply src/miner_lora.erl]: Added a comment

**Comment by Vagabond** [comment]
This was tested by a community member and it seems to have fixed his problem where his JSON had unexpected fields (because he was running a TTN packet forwarder).

**Comment by Vagabond** [comment]
I didnt mean to upgrade rebar3, will revert that.

**Pull request merged**